Source/CarthageKit/GitHub.swift

```diff
@@ -167,6 +167,18 @@ extension URLSession {
 		let configuration = URLSessionConfiguration.default
 		configuration.connectionProxyDictionary = Proxy.default.connectionProxyDictionary
 
-		return URLSession(configuration: configuration)
+		let delegate = GitHubURLSessionDelegate()
+		return URLSession(configuration: configuration, delegate: delegate, delegateQueue: OperationQueue.current)
+	}
+}
+
+class GitHubURLSessionDelegate: NSObject, URLSessionTaskDelegate {
+	func urlSession(_ session: URLSession, task: URLSessionTask, willPerformHTTPRedirection response: HTTPURLResponse, newRequest request: URLRequest, completionHandler: @escaping (URLRequest?) -> Void) {
+		let originalRequest = task.originalRequest
+		var newRequest = request
+		if let auth = originalRequest?.value(forHTTPHeaderField: "Authorization") {
+			newRequest.setValue(auth, forHTTPHeaderField: "Authorization")
+		}
```

Comment:
Can you add a check to only do this if you're redirected to the same domain?

I don't think it should matter, but I'd like to be conservative with peoples' credentials.

Reply:
Yeh good idea. I also added a couple of tests around this logic.